plugins/rolereactions.py

```diff
@@ -0,0 +1,213 @@
+import re
+import discord
+import discord_client
+import discord.utils
+import util.db.kv
+import util.discord
+import plugins.commands
+import plugins.privileges
+
+conf = util.db.kv.Config(__name__)
+msg_id_re = re.compile(
+    r"(?:https?://(?:\w*\.)?(?:discord.com|discordapp.com)"
+    r"/channels/\d+/\d+/)?(\d+)"
+)
+
+async def find_message(guild, msg_id):
+    for channel in guild.channels if guild else ():
+        if not isinstance(channel, discord.TextChannel): continue
+        try:
+            return await channel.fetch_message(msg_id)
```

Comment:
Each of these is an API call. Not a good idea when there's 100 channels.

Reply:
What else can I do when I just have the message id? I can try to get the message from the reference or channel id (in the link) otherwise though.

Reply:
Seems reasonable to just remove the choice of using message id.

Reply:
I now force that the user refer to the message using either a reference or a link (i.e. something that includes the channel id), and store that information: this enables the user to see links on which they have added role reactions.